esrally/track/loader.py

```diff
@@ -1076,6 +1085,17 @@ def read(self, track_name, track_spec_file, mapping_dir):
 
         current_track = self.read_track(track_name, track_spec, mapping_dir, track_spec_file)
 
+        internal_user_defined_track_params = self.complete_track_params.internal_user_defined_track_params()
+        if len(internal_user_defined_track_params) > 0:
+            err_msg = "Some of your track parameter(s) {} are defined by Rally and cannot be modified.\n".format(
+                ",".join(opts.double_quoted_list_of(sorted(internal_user_defined_track_params)))
+            )
+
+            self.logger.critical(err_msg)
```

Comment:
As discussed offline, here we might want to consider the recommendation in https://www.palkeo.com/en/blog/python-logging.html#why i.e. something like:

`self.logger.critical("Some of your track parameter(s) %s are defined by Rally and cannot be modified", join(opts.double_quoted_list_of(sorted(internal_user_defined_track_params))`

I understand this contradicts my earlier comment https://github.com/elastic/rally/pull/1750/files#r1266695052 (and it's not super critical). We generally use the above advice for logging statements in Rally.

Reply:
We use the same message twice - once for the logger, and once in the console output. To avoid repetition I've decided to use f-string as suggested above.